tutorials/working_with_auxiliary_data.md

```diff
@@ -0,0 +1,89 @@
+ ```Python
+from cornac.data import TextModality
+from cornac.data.text import BaseTokenizer
+
+item_text_modality = TextModality(corpus=plots, ids=movie_ids, 
+                                  tokenizer=BaseTokenizer(sep='\t', stop_words='english'),
+                                  max_vocab=5000, max_doc_freq=0.5)
+```
+In addition the movies plots and ids, we have specified a tokenizer to split text, we limited the maximum vocabulary size to 5000, as well as filtered out words occurring in more than 50% of documents (plots in our case) by setting `max_doc_freq = 0.5`. For more options/details please refer to the [docs](https://cornac.readthedocs.io/en/latest/data.html#module-cornac.data.text). 
+ 
+ 
+**Bag-of-Word text representation.** CDL assumes the bag-of-word representation for text information, i.e., in the form of a document-word matrix. The good news is that we don't have to worry about how to generate such a representation from our raw texts. The TextModality class implements the necessary routines to process and output different representations for text data, e.g., sequence, bag of word. That is, to get our auxiliary data under the desired format, all we need inside the CLD code `cornac/models/cdl/recom_cld` is the following line of code:
+```Python
+text_feature = self.train_set.item_text.batch_bow(np.arange(n_items))
+``` 
+where `self.train_set.item_text` is set to our instantiated text modality `item_text_modality`, `n_items` is number of training items, and the `batch_bow()` function returns the bag-of-word vectors of the specified item ids, in our case we want the text features for all training items: `np.arange(n_items)`.
+
+**Aligning auxiliary information with the set of training items.** Another important aspect worth mentioning at this level is that, we don't have to take extra actions to align the set of training movies with their plots. They are already aligned. That is, the first row in `text_feature` corresponds to training `movie_id: 1`, the second row to training `movie_id: 2`, and so on. This is made possible thanks to passing `item_text_modality` through the evaluation (splitting) method as we shall see shortly. 
+
+### Evaluation
+
+We are now left with the standard steps to fit and evaluate a recommender model using Cornac. We use train-test splitting as our evaluation method:
+```Python
+from cornac.eval_methods import RatioSplit
+
+ratio_split = RatioSplit(data=rating_data, test_size=0.2, exclude_unknowns=True,
+                         item_text=item_text_modality, verbose=True,
+                         seed=123, rating_threshold=0.5)
+``` 
+The text modality passed for the evaluation method. As mentioned earlier, this make it possible to avoid the tedious process of aligning the set of training items with their auxiliary data. Next, we need to instantiate the CLD model and evaluation metrics,
```

Comment:
The item text modality is passed into the evaluation method. As mentioned earlier, this makes it possible to avoid the tedious process of aligning the set of training items with their auxiliary data. Moving forward, we need to instantiate the CDL model and evaluation metrics: